Security Vulnerability in used spring-messaging-5.3.2.jar

**Type**: Bug / dependency-vulnerability

**groupId:** io.awspring.cloud
**artifactId:** spring-cloud-starter-aws-messaging
**version:** 2.3.2 (latest as of December 2021, downgrading does not solve the problem)

**Component**:
"messaging", "SNS", "SQS"

**Describe the bug**
spring-messaging-5.3.2.jar has a severe security vulnerability. 

For the Record, we tried following the security vulnerability link, but it oddly enough did not lead to a correct site with an option to report it (instead it leads to VMWare and something called Tanzu, pretty far from spring-cloud-aws Ref. https://tanzu.vmware.com/security).
Since this is a showstopper for our team making an important release with SNS & SQS features, our only option is reporting it here. _We humbly apologize for any inconvenience_. If there was **any** other option, e.g. a later release of spring-cloud-aws without this vulnerability, we would gladly take it. Please advice if there is. We have spent significant time finding one, without any success.

Running the following standard `Maven` command exposes the vulnerability.
`mvn -DdataDirectory=~/cve-data -DcveValidForHours=24 -C -Pdep-vuln-check,static-analysis -B clean install`

`[ERROR] Failed to execute goal org.owasp:dependency-check-maven:6.1.6:check (default) on project one: 
[ERROR] 
[ERROR] One or more dependencies were identified with vulnerabilities that have a CVSS score greater than or equal to '4,0': 
[ERROR] 
[ERROR] spring-messaging-5.3.2.jar: CVE-2021-22096, CVE-2021-22118
[ERROR] 
[ERROR] See the dependency-check report for more details.`


Published Vulnerabilities

CVE-2021-22096  

In Spring Framework versions 5.3.0 - 5.3.10, 5.2.0 - 5.2.17, and older unsupported versions, it is possible for a user to provide malicious input to cause the insertion of additional log entries.

NVD-CWE-Other

CVSSv2:

    Base Score: MEDIUM (4.0)
    Vector: /AV:N/AC:L/Au:S/C:N/I:P/A:N

CVSSv3:

    Base Score: MEDIUM (4.3)
    Vector: CVSS:3.1/AV:N/AC:L/PR:L/UI:N/S:U/C:N/I:L/A:N


References:

    CONFIRM - https://security.netapp.com/advisory/ntap-20211125-0005/
    MISC - https://tanzu.vmware.com/security/cve-2021-22096

Vulnerable Software & Versions: (show all)

    cpe:2.3:a:vmware:spring_framework:*:*:*:*:*:*:*:* versions from (including) 5.3.0; versions up to (including) 5.3.10
    ...

CVE-2021-22118  

In Spring Framework, versions 5.2.x prior to 5.2.15 and versions 5.3.x prior to 5.3.7, a WebFlux application is vulnerable to a privilege escalation: by (re)creating the temporary storage directory, a locally authenticated malicious user can read or modify files that have been uploaded to the WebFlux application, or overwrite arbitrary files with multipart request data.

CWE-269 Improper Privilege Management

CVSSv2:

    Base Score: MEDIUM (4.6)
    Vector: /AV:L/AC:L/Au:N/C:P/I:P/A:P

CVSSv3:

    Base Score: HIGH (7.8)
    Vector: CVSS:3.1/AV:L/AC:L/PR:L/UI:N/S:U/C:H/I:H/A:H


References:

    CONFIRM - https://security.netapp.com/advisory/ntap-20210713-0005/
    MISC - https://tanzu.vmware.com/security/cve-2021-22118
    MISC - https://www.oracle.com/security-alerts/cpuoct2021.html
    N/A - N/A

Vulnerable Software & Versions: (show all)

    cpe:2.3:a:vmware:spring_framework:*:*:*:*:*:*:*:* versions from (including) 5.3.0; versions up to (excluding) 5.3.7
    ...

--- a/pom.xml
+++ b/pom.xml
@@ -22,7 +22,7 @@
 	<parent>
 		<groupId>org.springframework.cloud</groupId>
 		<artifactId>spring-cloud-build</artifactId>
-		<version>3.0.0</version>
+		<version>3.0.5</version>
 		<relativePath/><!-- lookup parent from repository -->
 	</parent>
 
@@ -48,7 +48,7 @@
 		<javax-mail.version>1.5.5</javax-mail.version>
 		<maven-deploy-plugin.version>2.8.2</maven-deploy-plugin.version>
 		<javax.activation.version>1.2.0</javax.activation.version>
-		<spring-cloud-commons.version>3.0.0</spring-cloud-commons.version>
+		<spring-cloud-commons.version>3.0.4</spring-cloud-commons.version>
 		<spring-javaformat.version>0.0.25</spring-javaformat.version>
 	</properties>
 

